klippy/extras/probe.py

```diff
@@ -490,6 +490,8 @@ def start_probe(self, gcmd):
         # Perform automatic probing
         self.lift_speed = probe.get_probe_params(gcmd)['lift_speed']
         self.probe_offsets = probe.get_offsets()
+        if method == 'tap':
+            self.probe_offsets = (0., 0., 0.)
```

Comment:
Of course, this wont work correctly outside of initial testing.  I suspect the probe code will need to be refactored so that `probe_session.pull_probed_results()` returns more information (it should probably return something like: bed_x, bed_y, bed_z, probed_x, probed_y - that is, return the position the nozzle was at during the probe along with the estimated bed height if not nozzle was moved to some other position).

Reply:
Even if I like the idea, and I already refactored some code in attept to achive that.

The problem is: most of the code does not care about real probe offsets.
The bed_mesh care and it is in a sense that every move happens to "probe" coordinates not "toolhead/nozzle".
`_move_next()`
And so,
I can't really make it move to probe coordinates, before I probe at least one point to get probe offsets from pull_probed_results() in this case.
Or if we do toolhead probe at the requested points (like all other code does), it would shift the bed mesh with probe offsets.

So, idk, my initial attept was to return offsets from probe_session (so probe can return whatever is neccesary), but it looked ugly so I scraped it.

Maybe I'm missing something.